code/modules/projectiles/guns/ballistic/bows/_bow.dm

```diff
@@ -20,13 +20,29 @@
 	internal_magazine = TRUE
 	cartridge_wording = "arrow"
 	bolt_type = BOLT_TYPE_NO_BOLT
+	click_on_low_ammo = FALSE
+	must_hold_to_load = TRUE
 	/// whether the bow is drawn back
 	var/drawn = FALSE
 
 /obj/item/gun/ballistic/bow/update_icon_state()
 	. = ..()
 	icon_state = chambered ? "[base_icon_state]_[drawn ? "drawn" : "nocked"]" : "[base_icon_state]"
 
+/obj/item/gun/ballistic/bow/AltClick(mob/user)
+	if(chambered)
+		chambered.forceMove(drop_location())
+		magazine.get_round(keep = FALSE)
+
+		var/obj/item/ammo_casing/caseless/arrow/our_arrow = chambered
+		user.put_in_hands(our_arrow)
+
+		drawn = FALSE
+		chambered = null
+		update_appearance()
+		return
+	return ..()
```

Comment:
early return here instead, also isnull is faster *mumble mumble*
```suggestion
	if(isnull(chambered))
		return ..()

	chambered.forceMove(drop_location())
	magazine.get_round(keep = FALSE)

	var/obj/item/ammo_casing/caseless/arrow/our_arrow = chambered
	user.put_in_hands(our_arrow)

	drawn = FALSE
	chambered = null
	update_appearance()
```

Reply:
Sure thing boss